desktop/core/src/desktop/js/sql/sqlUtils.ts

```diff
@@ -267,24 +267,51 @@ export default {
     sqlReferenceProvider?: SqlReferenceProvider,
     forceAppendBacktick?: boolean
   ): Promise<string> => {
+    // Determine the quote character based on the connector dialect
+    let quoteChar =
+      (connector.dialect_properties && connector.dialect_properties.sql_identifier_quote) || '`';
+
+    // Use double quotes for Trino
+    if (connector.dialect === 'trino') {
+      quoteChar = '"';
+    }
+
     if (forceAppendBacktick) {
-      return identifier + '`';
+      return identifier + quoteChar;
     }
-    const quoteChar =
-      (connector.dialect_properties && connector.dialect_properties.sql_identifier_quote) || '`';
-    if (identifier.indexOf(quoteChar) === 0) {
+
+    // Check if identifier is already quoted
+    if (identifier.startsWith(quoteChar) && identifier.endsWith(quoteChar)) {
       return identifier;
     }
+    // Check for multi-part identifiers (e.g., catalog.schema)
+    if (identifier.includes('.')) {
+      // Split the identifier into parts and quote each part separately
+      return identifier
+        .split('.')
+        .map(part => {
+          // Quote each part if it is not already quoted
+          if (part.startsWith(quoteChar) && part.endsWith(quoteChar)) {
+            return part;
+          } else {
+            return `${quoteChar}${part}${quoteChar}`;
+          }
+        })
+        .join('.');
+    }
+
     const reservedKeywords = await (
       sqlReferenceProvider || sqlReferenceRepository
     ).getReservedKeywords(connector.dialect || 'generic');
-    if (reservedKeywords.has(identifier.toUpperCase())) {
-      return quoteChar + identifier + quoteChar;
-    }
 
-    if (!/^[A-Za-z][A-Za-z0-9_]*$/.test(identifier)) {
+    // Quote the identifier if it is a reserved keyword or not a valid SQL identifier
+    if (
+      reservedKeywords.has(identifier.toUpperCase()) ||
+      !/^[A-Za-z][A-Za-z0-9_]*$/.test(identifier)
```

Comment:
I think there are SQL dialects that allow _ as the first character as well (if we are talking about table names etc)

Reply:
@bjornalm  Yes, I agree, but if it starts with a "-", we are enclosing the name with either backticks or double quotes.

Reply:
So perhaps I'm misunderstanding, but we want to "Quote the identifier if it is a reserved keyword or not a valid SQL identifier"
`reservedKeywords.has(identifier.toUpperCase())` = reserved keywords 
`!/^[A-Za-z][A-Za-z0-9_]*$/.test(identifier)` = not a valid SQL identifier
`/^[A-Za-z][A-Za-z0-9_]*$/.test(identifier)` = a valid SQL identifier

But that regex will not allow underscore as the first character of a table name, which many dialects do support.
Perhaps we should use `^[A-Za-z_][A-Za-z0-9_]*`